apps/web/src/ApplicationReadyGuard.tsx

```diff
@@ -1,25 +1,64 @@
 import { Navigate, useLocation } from 'react-router-dom';
 import { type PropsWithChildren, useLayoutEffect } from 'react';
-import { useAuth, useEnvironment } from './hooks';
-import { isStudioRoute } from './studio/utils/routing';
+import { useAuth, useEnvironment, useMonitoring, useRoutes } from './hooks';
+import { ROUTES } from './constants/routes';
+import { IS_EE_AUTH_ENABLED } from './config/index';
 
 export function ApplicationReadyGuard({ children }: PropsWithChildren<{}>) {
+  useMonitoring();
   const location = useLocation();
-  const { isLoading: isLoadingAuth, inPublicRoute } = useAuth();
-  const { isLoading: isLoadingEnvironment } = useEnvironment();
+  const { inPublicRoute, inStudioRoute } = useRoutes();
+  const { isUserLoaded, isOrganizationLoaded, currentUser, currentOrganization } = useAuth();
+  const { isLoaded: isEnvironmentLoaded } = useEnvironment();
 
-  const isLoading = isStudioRoute(location.pathname) ? isLoadingAuth : isLoadingAuth && isLoadingEnvironment;
+  const isLoaded = isUserLoaded && isOrganizationLoaded && isEnvironmentLoaded;
 
-  // Clean up the skeleton loader when the app is ready
   useLayoutEffect(() => {
-    if (inPublicRoute || !isLoading) {
-      document.getElementById('loader')?.remove();
+    if (inPublicRoute || inStudioRoute || isLoaded) {
+      const el = document.getElementById('loader');
+
+      if (!el) {
+        return;
+      }
+
+      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+      // @ts-expect-error
+      if (window.isPlaywright) {
+        el.remove();
+
+        return;
+      }
+
+      el.addEventListener('transitionend', () => el.remove(), { once: true });
+
+      requestAnimationFrame(() => el.classList.add('fade-out'));
+    }
+  }, [inPublicRoute, inStudioRoute, isLoaded]);
```

Comment:
<!-- Suggestions propose improvements to the current subject. It's important to be explicit and clear on what is being suggested and why it is an improvement. Consider using patches and the blocking or non-blocking decorations to further communicate your intent.  -->
✏ **suggestion:** Move to its own hook that is less tightly coupled with the route scope‏